api/src/main/java/io/druid/data/input/impl/CSVParseSpec.java

```diff
@@ -80,23 +103,23 @@ public void verify(List<String> usedCols)
   @Override
   public Parser<String, Object> makeParser()
   {
-    return new CSVParser(Optional.fromNullable(listDelimiter), columns);
+    return new CSVParser(Optional.fromNullable(listDelimiter), columns, hasHeaderRow, skipHeaderRows);
```

Comment:
I found some more methods which receive an Optional as its parameter like ```TaskLockBox.tryLock()``` or ```OverlordResource.asLeaderWith()```. I think it would be better to fix these codes all together. I'll open an issue for it.